### ISSUE TITLE
GLTFLoader: Scene hierarchy not consistant between Chrome and Firefox.

### ISSUE THREAD

**Issue by pikilipita** [mrdoob/three.js#18774]
I just found out the order of children in a traverse 'traverse' loop in a (glb file):

```js
object.traverse( function ( child ) {
//to something with child
});
```

is different on Chrome and on Firefox.

Actually, on Chrome, the order seem to be random ( is different each time the page is refreshed)

Tested with a glb file exported from blender, 
on Chrome & Firefox  linux and window 10, with three 112

Is this a known issue or is there a work around for this?

using threejs 112  & GLTF loader as of today's version : 2020/02/28

**Comment by Mugen87** [comment]
Can you please demonstrate this issue with a live example?

`Object3D.traverse()` executes the callback in a fixed order by traversing the hierarchy top-down. If the order of execution varies, the scene graph is actually composed in a different way.

**Comment by pikilipita** [comment]
here's a live demo ( see in console )
https://lune.live/demos/glb-traverse/

and here are my results: ( Chrome on left, Firefox on right )
![traverse order](https://user-images.githubusercontent.com/1799792/75560684-ef2a6e00-5a45-11ea-8587-d9d134ef391b.png)

**Comment by Mugen87** [comment]
I've checked both parsed glTF scenes in Chrome and Firefox. Like I said before, the object hierarchy is different. `Object3D.traverse()` works as expected.

It seems `GLTFLoader` composes the scene differently depending on the used browser.

**Comment by pikilipita** [comment]
direct link to the file: [https://lune.live/demos/glb-traverse/assets/transported_blender_2-8_10_fix.glb](url)

the same hierarchy differences can be reproduced in this gltf viewer:
[https://gltf-viewer.donmccurdy.com/](url)

( I've renamed the issue title )

**Comment by donmccurdy** [comment]
Neat model! 🙂

This seems to be the same issue as https://github.com/mrdoob/three.js/issues/15561. I thought we'd fixed it, but I guess the conversation there went off in another direction... I'm going to close that issue and move the discussion here.

**Comment by takahirox** [comment]
Oh sorry I made PR #15587 including this fix but it had a response time impact and I didn't have time to think how to resolve, and ended up closing the PR so far.

I think it's good to first fix only this issue. IIRC I explained the root issue in #15661, not hard to fix.

**Comment by pikilipita** [comment]
Would you have a time estimation on when the fix will be applied?

**Comment by donmccurdy** [comment]
No, sorry. You may want to traverse the tree and sort the objects by name in the meantime.

**Comment by Mugen87** [comment]
Merging into #15561.

**Comment by Mugen87** [comment]
Fixed via #25058.
